tools/triage.sh

```diff
@@ -0,0 +1,205 @@
+# one xep changed: git fetch origin +refs/pull/1249/merge; git checkout FETCH_HEAD; ./tools/triage.sh 'XEP-0444: bla'
+
+set -euo pipefail
+
+pr_title="$1"
+shift
+
+pr_commit=""
+if [ $# -ge 1 ]
+then
+    pr_commit="$1"
+    shift
+else
+    pr_commit="HEAD"
+fi
+
+merge_base=""
+if [ $# -ge 1 ]
+then
+    merge_base="$1"
+    shift
+else
+    merge_base="$(git merge-base "$pr_commit" master)"
+fi
+
+xpath() {
+    set -euo pipefail
+    xmllint --nonet --noent --xpath "$2" "$1" --nowarning --dtdvalid xep.dtd
+}
+
+files_changed="$(git --no-pager diff --name-only "$pr_commit" "$merge_base")"
+
+set +e
+protoxep="$(echo "$files_changed" | grep -E '^inbox/[^.]+.xml$')"
+num_protoxeps=$(echo "$protoxep" | grep -v '^$' | wc -l)
+set -e
+
+if [ $num_protoxeps -ge 1 ]
+then
+    # we are dealing with a protoxep here
+    echo "tag: protoxep"
+    if [ $num_protoxeps -ne 1 ]
+    then
+        echo "tag: split-pr"
+        echo "error: multiple protoxeps cannot be created in 1 PR"
+        exit 1
+    fi
+    expected_title="ProtoXEP: $(xpath "$protoxep" '/xep/header/title/text()')"
+    if [ "$pr_title" != "$expected_title" ]
+    then
+        echo "tag: bad-title"
+        echo "error: expected title '$expected_title' got title '$pr_title'"
+        exit 1
+    fi
+    echo "tag: ready-to-merge"
+    echo "manual: be sure type '$(xpath "$protoxep" '/xep/header/type/text()')' is appropriate for XEP content"
+    exit 0
+fi
+
+# we are *not* dealing with a protoxep here
+set +e
+xeps_changed="$(echo "$files_changed" | grep -E '^xep-[0-9]{4}.xml$')"
+xep_likes_changed="$(echo "$files_changed" | grep -E '^xep-[^.]+.xml$')"
+num_xeps_changed=$(echo "$xeps_changed" | grep -v '^$' | wc -l)
+set -e
+
+if [ "$xeps_changed" != "$xep_likes_changed" ]
+then
+    echo "error: xep files created/changed but named incorrectly: $(echo "$xep_likes_changed" | tr '\n' ' ')"
+    exit 0
+fi
+
+if [ $num_xeps_changed -eq 0 ]
+then
+    # some tag for this? tooling changes?
+    echo "manual: no XEPs changed"
+    exit 0
+fi
+
+if [ $num_xeps_changed -gt 1 ]
+then
+    echo "manual: multiple XEPs changed, split?"
+fi
+
+all_approvers=""
+# a super approver can approve this entire PR, because they are author on all XEPs touched
+super_approvers=""
+for xep in $xeps_changed
+do
+    xep_num="XEP-$(echo "$xep" | grep -Eo '[0-9]{4}')"
+    if ! echo "$pr_title" | grep "$xep_num" &>/dev/null
+    then
+        echo "error: title did not include '$xep_num'"
+        exit 1
+    fi
+    
+    status="$(xpath "$xep" '/xep/header/status/text()')"    
+    approvers=""
+    if echo "$status" | grep -E '^(Experimental|Deferred)$' &>/dev/null
+    then
+        # Experimental or Deferred
+        if [ "$status" == 'Deferred' ]
+        then
+            echo "tag: needs-editor-action"
+            echo "manual: move $xep_num status to Experimental when merged"
```

Comment:
The status change should happen in the same commit which writes the revision block.

Reply:
Currently this doesn't go by commit, it looks at the entire PR as 1 change :/

Reply:
Right, I meant that this should be part of the note for the editor. "when merged" can easily be read as "change status after merging" which would not be the correct action.

Reply:
I changed this language, let me know